src/lm.jl

```diff
@@ -156,7 +156,20 @@ function coeftable(mm::LinearModel)
               ["x$i" for i = 1:size(mm.pp.X, 2)], 4)
 end
 
-predict(mm::LinearModel, newx::Matrix) =  newx * coef(mm)
+predict(mm::LinearModel, newx::AbstractMatrix) = newx * coef(mm)
+
+function predict(mm::LinearModel, newx::AbstractMatrix, interval_type::Symbol, level = 0.95)
+    retmean = newx * coef(mm)
+    interval_type == :confint || error("only :confint is currently implemented") #:predint will be implemented
+
+    R = cholfact!(mm.pp)[:U] #get the R matrix from the QR factorization
+    residvar = (ones(size(newx,2),1) * deviance(mm)/dof_residual(mm))
+    retvariance = (newx/R).^2 * residvar
+
+    interval = quantile(TDist(dof_residual(mm)), (1 - level)/2) * sqrt(retvariance)
```

Comment:
np. For the `level` thing, we could make it a keyword argument and also supply `alpha = 1-level`, that would make it clear and allow users use the way they are most familiar with? @ararslan 

About the `sqrt.` I had that in the code earlier but it produced an error. I must have done something else wrong back then, as it works now :-)